src/main/java/com/slashswapper/SlashSwapperPlugin.java

```diff
@@ -57,23 +60,28 @@ public void onScriptCallbackEvent(ScriptCallbackEvent scriptCallbackEvent)
 		final int[] intStack = client.getIntStack();
 		int intStackCount = client.getIntStackSize();
 
-		// intStack[intStackCount -1] is the check to see if a user typed /,// or /// from non fc/cc/gc selected channels
-		// intStack[intStackCount - 2] is the ID to see what channel the message is routed into
+		// the check to see if a user typed /, // or /// from non fc/cc/gc selected channels
+		int slashCount = intStack[intStackCount - 1];
 
-		if (intStack[intStackCount - 1] == 1 && intStack[intStackCount - 2] == 9)
+		// the ID to see what channel the message is routed into
+		int targetChannelCode = intStack[intStackCount - 2];
+
+		// message is going to friends chat because user typed /, swap it to guest or clan chat
+		if (slashCount == 1 && targetChannelCode == FRIEND_CHAT)
 		{
-			// message is going to friends chat because user typed /
-			intStack[intStackCount - 2] = !swapGuestChat ? 41 : 44; // swap it to clan chat or guest chat
+			intStack[intStackCount - 2] = swapGuestChat ? GUEST_CHAT : CLAN_OR_GIM_CHAT;
 		}
-		else if (intStack[intStackCount - 1] == 2 && intStack[intStackCount - 2] == 41)
+
+		// message is going to clan chat because user typed //, swap it to friends chat
+		if (slashCount == 2 && targetChannelCode == CLAN_OR_GIM_CHAT)
 		{
-			// message is going to clan chat because user typed //
-			intStack[intStackCount - 2] = 9; // swap it to friends chat
+			intStack[intStackCount - 2] = FRIEND_CHAT;
 		}
-		else if (intStack[intStackCount - 1] == 3  && intStack[intStackCount - 2] == 44)
+
+		// message is going to guest chat because user typed ///, swap it to clan chat
+		if (slashCount == 3 && targetChannelCode == GUEST_CHAT && swapGuestChat)
 		{
-			// message is going to guest chat because user typed ///
-			intStack[intStackCount - 2] = !swapGuestChat ? 9 : 41; // swap it to friends chat or guest chat
+			intStack[intStackCount - 2] = CLAN_OR_GIM_CHAT;
 		}
```

Comment:
This part in particular. The old `!swapGuestChat ? 9 : 41` caused messages meant for Guest Chat (44) to be sent to Friends Chat (9) when the config is disabled. I changed it to always swap to CC (41) and I added `&& swapGuestChat` in the if-statement to only swap when the config is enabled.